app/controllers/bulkrax/importers_controller.rb

```diff
@@ -77,9 +77,14 @@ def create
       cloud_files = params.delete(:selected_files)
       @importer = Importer.new(importer_params)
       field_mapping_params
+      @importer.validate_only = true if params[:commit] == 'Create and Validate'
       if @importer.save
         files_for_import(file, cloud_files)
-        Bulkrax::ImporterJob.perform_later(@importer.id) if params[:commit] == 'Create and Import'
+        if params[:commit] == 'Create and Import'
+          Bulkrax::ImporterJob.perform_later(@importer.id)
+        elsif params[:commit] == 'Create and Validate'
+          Bulkrax::ImporterJob.perform_now(@importer.id)
+        end
         if api_request?
```

Comment:
@geekscruff Would refactoring this to use `@importer.parser.perform_method` work? If so, it would nix the need for the conditional entirely. 
E.g. 
```ruby
      if @importer.save
        files_for_import(file, cloud_files)
        Bulkrax::ImporterJob.send(@importer.parser.perform_method, @importer.id)
        if api_request?
```

Reply:
yes!